sdks/java/io/iceberg/src/main/java/org/apache/beam/sdk/io/iceberg/IcebergIO.java

```diff
@@ -37,14 +39,243 @@
+ *       Generally, a higher value will produce fewer, larger data files.
+ *       </td>
+ *   </tr>
+ *   <tr>
+ *     <td> {@code catalog_name} </td> <td> {@code str} </td> <td> The name of the catalog. Defaults to {@code apache-beam-<VERSION>}. </td>
+ *   </tr>
+ *   <tr>
+ *     <td> {@code catalog_properties} </td> <td> {@code map<str, str>} </td> <td> A map of properties to be used when
+ *     constructing the Iceberg catalog. Required properties will depend on what catalog you are using, but
+ *     <a href="https://iceberg.apache.org/docs/latest/configuration/#catalog-properties">this list</a>
+ *     is a good starting point. </td>
+ *   </tr>
+ *   <tr>
+ *     <td> {@code config_properties} </td> <td> {@code map<str, str>} </td> <td> A map of properties
+ *     to instantiate the catalog's Hadoop {@link Configuration}. Required properties will depend on your catalog
+ *     implementation, but <a href="https://iceberg.apache.org/docs/latest/configuration/#hadoop-configuration">this list</a>
+ *     is a good starting point.
+ *   </tr>
+ * </table>
+ *
+ * <p><b>Additional configuration options are provided in the `Pre-filtering Options` section below,
+ * for Iceberg writes.</b>
+ *
+ * <h3>Beam Rows</h3>
+ *
+ * <p>Being a Managed transform, this IO exclusively writes and reads using Beam {@link Row}s.
+ * Conversion takes place between Beam {@link Row}s and Iceberg {@link Record}s using helper methods
+ * in {@link IcebergUtils}. Below is a type conversion table mapping Beam and Iceberg types:
+ *
+ * <table border="1" cellspacing="1">
+ *   <tr>
+ *     <td> <b>Beam {@link Schema.FieldType}</b> </td> <td> <b>Iceberg {@link Type}</b>
+ *   </tr>
+ *   <tr>
+ *     <td> BYTES </td> <td> BINARY </td>
+ *   </tr>
+ *   <tr>
+ *     <td> BOOLEAN </td> <td> BOOLEAN </td>
+ *   </tr>
+ *   <tr>
+ *     <td> STRING </td> <td> STRING </td>
+ *   </tr>
+ *   <tr>
+ *     <td> INT32 </td> <td> INTEGER </td>
+ *   </tr>
+ *   <tr>
+ *     <td> INT64 </td> <td> LONG </td>
+ *   </tr>
+ *   <tr>
+ *     <td> DECIMAL </td> <td> STRING </td>
+ *   </tr>
+ *   <tr>
+ *     <td> FLOAT </td> <td> FLOAT </td>
+ *   </tr>
+ *   <tr>
+ *     <td> DOUBLE </td> <td> DOUBLE </td>
+ *   </tr>
+ *   <tr>
+ *     <td> DATETIME </td> <td> STRING </td>
+ *   </tr>
+ *   <tr>
+ *     <td> ITERABLE </td> <td> LIST </td>
+ *   </tr>
+ *   <tr>
+ *     <td> ARRAY </td> <td> LIST </td>
+ *   </tr>
+ *   <tr>
+ *     <td> MAP </td> <td> MAP </td>
+ *   </tr>
+ *   <tr>
+ *     <td> ROW </td> <td> STRUCT </td>
+ *   </tr>
+ * </table>
+ *
+ * <h3>Dynamic Destinations</h3>
+ *
+ * <p>Managed Iceberg supports writing to dynamic destinations. To do so, please provide an
+ * identifier template for the <b>{@code table}</b> parameter. A template should have placeholders
+ * represented as curly braces containing a record field name, e.g.: {@code
+ * "my_namespace.my_{foo}_table"}.
```

Comment:
Curly braces are not valid characters for actual table names right ? Just to make sure that we do not end up trying to parse valid table names as schema fields.

Reply:
Yep, trying to create a table with curly braces results in an error 400 `Invalid table ID`